craft_parts/plugins/base.py

```diff
@@ -62,6 +62,11 @@ def get_build_environment(self) -> Dict[str, str]:
 
     @property
```

Comment:
Can I remove the `out_of_source_build` property?

The only place it is used across all craft libraries and *craft applications is the `ros` plugin in snapcraft, which I would like to update.  It'd be a 1-line fix.